File: src/main/java/com/github/konbluesky/jdph/listeners/MyProjectManagerListener.java
package com.github.konbluesky.jdph.listeners;

import com.github.konbluesky.jdph.setting.AppSettingsState;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.project.ProjectManagerListener;
import org.jetbrains.annotations.NotNull;

public class MyProjectManagerListener implements ProjectManagerListener {

    @Override
    public void projectOpened(@NotNull Project project) {
    }
}

File: src/main/kotlin/com/github/konbluesky/javadesignpatternhelperideaplugin/listeners/MyProjectManagerListener.kt
package com.github.konbluesky.javadesignpatternhelperideaplugin.listeners

import com.github.konbluesky.javadesignpatternhelperideaplugin.services.MyProjectService
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project
import com.intellij.openapi.project.ProjectManagerListener

internal class MyProjectManagerListener : ProjectManagerListener {

    override fun projectOpened(project: Project) {
        project.service<MyProjectService>()
    }
}
